examples/applications/plot_1d_constrained_example.py

```diff
@@ -0,0 +1,104 @@
+# Next, we simulate a standard de-noising setting by contaminating a known
+# signal :math:`x`` with some Gaussianly distributed noise. Note that for simplicity the
+# measurement operator here is taken to be the identity operator.
+
+size = 2048                                              # Dimension of the 1D vector
+ISNR = 20.0                                              # Input signal to noise ratio
+sigma = 10 ** (-ISNR / 20.0)                             # Noise standard deviation
+epsilon = np.sqrt(size + 2.0 * np.sqrt(size)) * sigma    # Radius of l2-ball
+
+x = normal_dist(0, 0.5).pdf(np.linspace(-2, 2, size))    # Ground truth signal x
+y = x + np.random.normal(0, sigma, size)                 # Simulated observations y
+
+##############################################################################
+# For the constrained problem with Gaussian noise the data-fidelity constraint
+# is given by a projection onto the :math:`\ell_2`-ball. Here we set up a linear-operator
+# corresponding to a forward and adjoint projection onto the :math:`\ell_2`-ball.
+
+p = prox_operators.l2_ball(epsilon, y, linear_operators.identity())
+p.beta = 1.0
+
+##############################################################################
+# We regularise this inverse problem by adopting a wavelet sparsity :math:`\ell_1`-norm prior.
+# To do this we first define what wavelets we wish to use, in this case a
+# combination of Daubechies family wavelets, and which levels to consider.
+# Any combination of wavelet families available by the pywavelet package may be
+# selected -- see https://tinyurl.com/5n7wzpmb
+
+wav = ["db1", "db4", "db6"]                               # Wavelet dictionaries to combine
+levels = 6                                                # Wavelet levels to consider [1-6]
+shape = (size,)                                           # Shape of nd-wavelets
+psi = linear_operators.dictionary(wav, levels, shape)     # Wavelet linear operator
+
+##############################################################################
+# Next we construct the :math:`\ell_1`-norm proximal operator which we pass the wavelets
+# (:math:`\Psi`) as a dictionary in which to compute the :math:`\ell_1`-norm. We also add an
+# additional reality constraint f for good measure, as we know a priori our
+# signal :math:`x` is real.
+
+h = prox_operators.l1_norm(np.max(np.abs(psi.dir_op(y))) * 1e-2, psi)
+h.beta = 1.0
+f = prox_operators.real_prox()
+
+##############################################################################
+# Finally we run the optimisation...
+
+best_estimate, diagnostics = primal_dual.FBPD(y, options, None, f, h, p, None)
+
+##############################################################################
+# ...and plot the results!
+
+plt.plot(np.real(y), "o", markersize=1)
+plt.plot(np.real(x), linewidth=2)
+plt.plot(np.real(best_estimate), linewidth=2)
+plt.legend(["data", "true", "fit"])
+
```

Comment:
Maybe add some labels to the axes

Reply:
I think perhaps this is redundant? it's basically just x and f(x)